modules/reactor/include/meshgenerators/AssemblyMeshGenerator.h

```diff
@@ -88,6 +88,21 @@ class AssemblyMeshGenerator : public ReactorGeometryMeshBuilderBase
   ///The name of the assembly outer boundary, equal to the concatenation of "outer_assembly_" and the assembly-type ID
   std::string _assembly_boundary_name;
 
+  /// Metadata of the maximum radius of ring regions
+  Real & _max_radius_meta;
+
+  /// MeshMetaData: number of radial intervals of the background region
+  unsigned int & _background_intervals_meta;
+
+  /// MeshMetaData: maximum node id of the background region
+  dof_id_type & _node_id_background_meta;
+
+  /// MeshMetaData: pitch size to be used when stitching with assembly meshes
+  Real & _pattern_pitch_meta;
+
+  /// MeshMetaData: azimuthal angles of all nodes
+  std::vector<Real> & _azimuthal_angle_meta;
+
```

Comment:
do they need to be class attributes?
You are not using them anywhere. 

While I agree it's been done a lot so far, maybe we should reconsider storing twice all metadata as attributes

Reply:
I m ticking now because some of these are a little odd. The max ring radius isnt typically what you care about for an assembly, it s very tied to a downstream application, separate from this object. Same for the azimuthal angles

Reply:
Looked into the underlying submeshgenerator code a bit more closely, the only metadata that actually needs to be transferred is max_radius_meta from PinMeshGenerator (defined in PCCMG) to AssemblyMeshGenerator (used by PatternedHexMeshGenerator), and pattern_pitch_meta from AssemblyMeshGenerator (defined in PatternedHexMeshGenerator) to CoreMeshGenerator (used by PatternedHexMeshGenerator) . The rest of the metadata isn't actually used anywhere for the specific workflow of Pin MG -> Assembly MG -> Core MG. However, these metadata need to be transferred explicitly in PinMeshGenerator and AssemblyMeshGenerator since there are intermediate mesh generators (such as BoundaryDeletionGenerator) between Pin MG, Assembly MG, and Core MG, an metadata is not transferred between these intermediate mesh generators.

With this in mind, is the preferred approach just to define class attirbutes for max_radius_meta in PinMG and pattern_pitch_meta in AssemblyMG and delete all the other class attributes that were defined in this PR?

Reply:
why use a class attribute for these two (max_radius_meta, pattern_pitch_meta) even? Cant you retrieve them from the metadata after it s been declared?

I understand all these metadata need to be passed, I m not sure why they need to be stored in the class

@miaoyinb is working on a cleaner metadata forwarding for the future btw

Reply:
Currently ‘declareMeshProperty’ can’t be called outside of the constructor class, otherwise Moose throws an error. Therefore the workaround, which is a bit long-winded, is to store the reference to declareMeshProperty as a class variable and then update the class variable once the variable value is available in generate()

Reply:
why dont we just add a getter that lets you modify the metadata or a setter

Reply:
@permcody @roystgnr thoughts?

Reply:
Just wondering, what would the getter or setter look like? Feels like it would be very similar to declareMeshProperty without the constraint for restricting the call to constructors only

Reply:
declareMeshProperty would stay in the constructor. 
However, the setter would:
- only work if the property has been declared already
- modify the data
There's already a getter. It's just const so it cant be used to modify

Reply:
@GiudGiud I am running into an issue with updating the code according to the suggestions. I am creating a function `setMeshProperty` that can be called after the constructor, assuming that `declareMeshProperty` has been called for the metadata of interest. However, I am running into an error in the function `MooseApp::registerRestartableData`, since it doesn't allow you to modify the metadata after it has been declared (see below in the `if data.declared` block):
<img width="784" alt="image" src="https://user-images.githubusercontent.com/14334316/210387489-2154fd17-624d-4fc3-9931-e34a37872f0f.png">

Any ideas how I should bypass this? Not sure if I should be modifying `MooseApp::registerRestartableData`

Reply:
I dont think you should call registerRestartableData again if it s been called in declareMeshProperty.

Are you doing this to get a reference to the restartable data?
Or to "save" the data in _restartable_data ?

Reply:
Thanks, I was calling `registerRestartableData` but now I am getting the reference to the restartable data directly from RestartableDataMap, can you check the `setMeshProperty` function now and let me know if this is what you had in mind? If so, I can address the remaining comments

Reply:
Removed all these class variables now